### ISSUE TITLE
Cloak of Shadows not working correctly

### ISSUE THREAD

**Issue by elasdmehico** [SunwellTracker/issues#1899]
Decription: Instantly removes all existing harmful spell effects and increases your chance to resist all spells by 90% for 5 sec. Does not remove effects that prevent you from using Cloak of Shadows.

How it works: It makes you immune to chains of ice.

How it should work: It should not make you immune to chains of ice.

Source (you should point out proofs of your report, please give us some source): https://www.wowhead.com/forums&topic=125361

"Hi wowhead forumers. I was wondering if anyone could tell me whether or not Chain of Ice and Deathgrip ignores the use of "Cloak of Shadows"? I was doing some world pvp against a couple of DK's on my rogue and I got hit and slowed every single time I used CoS. "

https://www.wowhead.com/forums&topic=150477
"Also frost fever applications from chains of ice go through cloak of shadows and grounding totems."

**Comment by Artekd** [comment]
Also, it shouldn't make the rogue immune from applying blood plague, because plague strike is a physical ability and should ignore CoS
http://www.arenajunkies.com/topic/98243-rogue-is-clos-brokenbugged/
https://us.battle.net/forums/en/wow/topic/9753835397

**Comment by elasdmehico** [comment]
bump

**Comment by Azathothh** [comment]
@artekd 
you're saying that it's bugged or what do you mean?
![obraz](https://user-images.githubusercontent.com/37483678/47714293-ca996880-dc3c-11e8-82da-5dc76eecf72b.png)
CoS isn't working on plague strike now and it's correct behaviour.
Death grip is physical ability and should go through CoS as well and it does.

CoS doesn't make you "immune" to anything, what it does is "Increases chance to resist spells by 90%." , so please don't use term "immune" cause it may be misleading for some players reading bugtracker.

"How it works: It makes you immune to chains of ice." - like said above - it's not true - CoS increase chance to resist CoI by 90%
"How it should work: It should not make you immune to chains of ice." - not true as well - CoI should be affected by CoS

Only thing that is working here different than on retail is Frost Fever not being applied when CoI doesn't land because of CoS. But i don't know if it was like that on retail all the time or if it was fixed.

http://www.arenajunkies.com/topic/135372-333-bug-chains-of-ices-disease-application-goes-through-grounding-and-cloak/

**Comment by elasdmehico** [comment]
"Frost Fever and Ebon Plaguebringer will now be applied through Grounding Totem and Cloak of Shadows by Chains of Ice 100% of the time. "
This is intended and they did this in cataclysm too. Cloaks miss chance shouldn't affect chains of ice, not the slow nor the dot.

I think the reason why chains should go through cloak 100% of the time is because it's considered an ability, not a spell.

**Comment by Azathothh** [comment]
frost fever =/= chains of ice 
"I think the reason why chains should go through cloak 100% of the time is because it's considered an ability, not a spell." where did you get that info from?

**Comment by elasdmehico** [comment]
https://www.wowhead.com/forums&topic=125361

"Hi wowhead forumers. I was wondering if anyone could tell me whether or not Chain of Ice and Deathgrip ignores the use of "Cloak of Shadows"? I was doing some world pvp against a couple of DK's on my rogue and I got hit and slowed every single time I used CoS. "

https://www.wowhead.com/forums&topic=150477
"Also frost fever applications from chains of ice go through cloak of shadows and grounding totems."
